pkg/client/client_venafi_cloud.go

```diff
@@ -260,13 +261,20 @@ func (c *VenafiCloudClient) Post(path string, body io.Reader) (*http.Response, e
 		return nil, err
 	}
 
-	req, err := http.NewRequest(http.MethodPost, fullURL(c.baseURL, path), body)
+	encodedBody := &bytes.Buffer{}
+	gz := gzip.NewWriter(encodedBody)
```

Comment:
I thought that compression is handled automatically by the Go http client: https://cs.opensource.google/go/go/+/refs/tags/go1.23.2:src/net/http/transport.go;l=194-202

Reply:
I did some searching and could not find an automatic implementation for http requests on the client side. This seems to be implemented only for responses and when receiving requests.